clang/include/clang/Basic/DiagnosticASTKinds.td

```diff
@@ -96,6 +96,8 @@ def note_constexpr_pointer_constant_comparison : Note<
   "at runtime">;
 def note_constexpr_literal_comparison : Note<
   "comparison of addresses of literals has unspecified value">;
+def note_constexpr_opaque_call_comparison : Note<
+  "comparison against opaque constant has unspecified value">;
```

Comment:
I can't imagine a normal person knows what's meant by "opaque constant"

Reply:
Yeah, opaque constant is pretty opaque, but I'm struggling to think of a better way to phrase this. It only triggers when calling a builtin which forms an address constant.

Mayyyybbbe: comparison against an anonymous object with a constant address has unspecified value

? (or maybe 'unnamed' instead of 'anonymous')

Reply:
It's not necessarily anonymous, or an object. For example, we get one of these constants from `__builtin_start_address(function)`, where the returned address is often the address of the function (though in some cases it might be a different code address -- and we don't know).

Added the actual pointer value to the diagnostic:
```
note: comparison against opaque constant address '&__builtin_function_start(a)' has unspecified value
```
Hopefully that makes it a bit clearer what the problem is :)